server/src/session.ts

```diff
@@ -131,6 +131,51 @@ export class Session {
     // project as dirty to force update the graph.
     project.markAsDirty();
     this.info(`Enabling Ivy language service for ${project.projectName}.`);
+
+    // Send diagnostics since we skipped this step when opening the file
+    // (because language service was disabled while waiting for ngcc).
+    // Technically we could just call project.refreshDiagnostics(), but we need
+    // to handle the special case where all open files are external templates.
+    const openFiles = this.getOpenFilesForNewlyLoadedProject(project);
+    this.sendPendingDiagnostics(openFiles);
+  }
+
+  /**
+   * Get the open files for a newly loaded project. If all the open files are
+   * external templates, then try to match them to the specified `project` by
+   * including one of the root files. The inclusion of a root file triggers
+   * global analysis, during which we fix up the external templates.
+   */
+  private getOpenFilesForNewlyLoadedProject(project: ts.server.Project): string[] {
+    const openFiles: string[] = [];
+
+    const iterable = this.projectService.openFiles.keys();
+    for (let it = iterable.next(); !it.done; it = iterable.next()) {
```

Comment:
Does `for..of` not work here?

Reply:
This is [TS-internal iterator](https://github.com/microsoft/TypeScript/blob/eac0f6dc3085647ce301bb79516cd925e528911c/src/compiler/corePublic.ts#L90), which is different from the ES 2015 iterator protocol. It does not have the `iterator` symbol, so `for..of` doesn't work. I'm not quite sure why TS needs its own iterator interface .. perhaps for compatibility with older runtimes that do not support ES 2015? 🤷‍♂️ 
They even have their own `arrayFrom`, but it's an internal method:
https://github.com/microsoft/TypeScript/blob/eac0f6dc3085647ce301bb79516cd925e528911c/src/compiler/core.ts#L1329-L1337

Reply:
Odd, thanks for the details!